### PR TITLE
hibernate_1

### PR DESCRIPTION
**Pull request by yurwing**
created hibernate.cfg.xml, HibernateUtil, model, dao and service for Movie, DataProcessingException

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            transaction.commit();
+            movie.setId(itemId);
+            return movie;
+        } catch (DataProcessingException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot add movie " + movie, e);
+        } finally {
+            session.close();
```
Comment (oroboro-cpu) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: Add a check for null also

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (oroboro-cpu) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: You don’t need to set id for your entity explicitly it’ll be done by Hibernate

**Review by oroboro-cpu — CHANGES_REQUESTED** [review]

**Review by oroboro-cpu — APPROVED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> all = session.createQuery("from Movie", Movie.class);
+            return all.getResultList();
+        } catch (DataProcessingException e) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: same

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (DataProcessingException e) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: where is it thrown in try block?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**